PrimeDAO/prime-deals-dapp
# 970 - Deepdao integration cl web comp

## Note:
### To test with Firebase Emulator- 
+ Add an `.env` file to the `FirebaseFunctions` functions (see Notion)
+ Uncomment the `callDeepDaoAPI` function in `firebaseFunctions/src/index.ts` in order to be able to invoke it later in the emulator (make sure to comment out again after deployment, so it is not pushed to production)
+ Deploy the firebase function: `npm run firebase-deploy-deepdao-function:default`
+ When done, invoke it from the browser: http://localhost:5001/prime-deals-6ace4/us-central1/callDeepDaoAPI
This will create a new `deep-dao` collection in the emulator with filled with data.
If running using Firebase cloud- the collection is already live.

+ run `npm ci` to install the new package `cl-webcomp-poc`.

## What was done
### Firebase
- a Firebase function pulls every 24 hours the DAO's list from DeepDAO API and maps it into the `deep-dao` collection in Firestore.
- This collection consists of (5) tranks of records containing minimal DAO data:
```JSON
{
  "<deep-dao-org-id>": {
    "avatarUrl": "<https://.../avatar.png>",
    "name": "<Dao Name>"
    "treasuryAddresses": [
      "<0x...1>", "<0x...2>",
    ]
  }
}
```

### dApp

![image](https://user-images.githubusercontent.com/2517870/185386943-e0876de6-6588-4ab1-9d7a-6f5649105093.png)

![image](https://user-images.githubusercontent.com/2517870/185386995-b66d8ebc-a54b-45c7-8cc3-a063a830a957.png)

![image](https://user-images.githubusercontent.com/2517870/185387029-19235207-e1f7-4add-b3df-685397ca3ff9.png)

![image](https://user-images.githubusercontent.com/2517870/185387085-f6f49292-2861-412a-9748-940f2074b410.png)

![image](https://user-images.githubusercontent.com/2517870/185387172-026e85db-569f-4769-af6f-da5a08a0f54d.png)
